lib/collection/src/operations/config_diff.rs

```diff
@@ -85,6 +86,79 @@ pub struct HnswConfigDiff {
     pub payload_m: Option<usize>,
 }
 
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Validate, Clone, Merge)]
+pub struct StrictModeConfig {
+    // Global
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub enabled: Option<bool>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub max_filter_limit: Option<usize>,
```

Comment:
I didn't call it 'limit' so we can have different limits for different contexts in future.
I agree with `filter_limit` sounding a bit confusing but since we're sharing this with eg. discovery, `search_limit` might also be confusing as well.
What do you think?

Reply:
`max_query_limit`? And specify the apis that it applies to